guix-science/packages/geant4.scm

```diff
@@ -0,0 +1,699 @@
+       ;; no tests in Makefile
+       #:tests? #f))
+    (native-inputs `(("G4NDL" ,g4ndl-4.7)
+                     ("G4EMLOW" ,g4emlow-8.2)
+                     ("G4PhotonEvaporation" ,photon-evaporation-5.7)
+                     ("G4RadioactiveDecay" ,radioactive-decay-5.6)
+                     ("G4PARTICLEXS" ,g4particlexs-4.0)
+                     ("G4PII" ,g4pii-1.3)
+                     ("G4RealSurface" ,real-surface-2.2)
+                     ("G4SAIDDATA" ,g4saiddata-2.0)
+                     ("G4ABLA" ,g4abla-3.1)
+                     ("G4INCL" ,g4incl-1.0)
+                     ("G4ENSDFSTATE" ,g4ensdfstate-2.3)
+                     ("G4TENDL" ,g4tendl-1.4)))
+    (home-page "https://geant4.web.cern.ch")
+    (synopsis "Monte Carlo particle track simulations")
+    (description
+     "Geant4 is a toolkit for the simulation of the passage of particles
+through matter.  Its areas of application include high energy,
+nuclear and accelerator physics, as well as studies
+in medical and space science.
+
+Note this package does not support visualisation -- you
+can use @code{geant4-vis} for that.")
+    (license (license:non-copyleft
+              "https://geant4.web.cern.ch/download/license"))))
+
+(define-public geant4-11-1-2
+  (package
+    (inherit geant4-11-1-1)
+    (version "11.1.2")
+    (source
+     (origin
+       (inherit (package-source geant4-11-1-1))
+       (uri (git-reference
+             (url "https://gitlab.cern.ch/geant4/geant4")
+             (commit (string-append "v" version))))
+       (sha256
+        (base32 "03dbq48sycw8bfsf2vm6kk9f7ri663lfiw3blwigck0g4f1vdmhd"))))))
+
+(define-public geant4-11-1-3
+  (package
+    (inherit geant4-11-1-1)
+    (version "11.1.3")
+    (source
+     (origin
+       (inherit (package-source geant4-11-1-1))
+       (uri (git-reference
+             (url "https://gitlab.cern.ch/geant4/geant4")
+             (commit (string-append "v" version))))
+       (sha256
+        (base32 "0w616j4w8gmmn5815pnyylm9sv0gpdicpjbvxwpyhzqlssmm2bjl"))))))
+
+;; -- geant4-vis: Geant4 with visualisation using OGL + Qt -- 
+
+;; (propagated-inputs (list cmake gnu-make gcc-toolchain mesa))
+(define-public geant4-vis-11-1-1
+  (package
+    (inherit geant4-11-1-1)
+    (name "geant4-vis")
+    (inputs (list coreutils
+                  gcc-toolchain
+                  xerces-c
+                  expat
+                  clhep-2.4.6.2
+                  python-2
+                  python-3.10
+                  perl
+                  tcsh
+                  qtbase-5
+                  libxmu
+                  libxt))
+    (arguments
+     `(#:configure-flags (let* ((out (assoc-ref %outputs "out"))
+                                (qt-path (string-append (assoc-ref
+                                                         %build-inputs
+                                                         "qtbase")
+                                                        "/lib/cmake/Qt5")))
```

Comment:
Same as above.  Please use G-expressions and replace `(assoc-ref %outputs "out")` with `#$output`, and `(assoc-ref %build-inputs ...)` with `#$(this-package-input ...)`.